# Identify missing datasets

In [1]:
## import required packages
import numpy as np
import matplotlib.pyplot as plt
import xarray as xr
import s3fs
import pandas as pd
from datetime import datetime
import json

### Generate list of dates for complete dataset

In [2]:
## create list of all expected dates for a complete daily mean dataset
dates = pd.date_range(start="2014-01-15",end="2021-02-07").tolist()
date_list_complete = []
for i in range(len(dates)):
    date_list_complete.append(str(dates[i])[:10])

In [3]:
len(date_list_complete)

2581

For a complete dataset, there should be 2581 netCDFs, one for each day

### Create dictionary for missing dates for each dataset using processed datasets (netCDF filenames)

In [4]:
def find_missing_dates(date_list_complete, file_dates_list):
    return [string for string in date_list_complete if string not in file_dates_list]

In [5]:
def identify_missing_datasets(variables_list, s3_netcdf_dir, sassie_s3 = True):

    ## create list of all expected dates for a complete daily mean dataset
    dates = pd.date_range(start="2014-01-15",end="2021-02-07").tolist()
    date_list_complete = []
    for i in range(len(dates)):
        date_list_complete.append(str(dates[i])[:10])

    if sassie_s3:
        s3 = s3fs.S3FileSystem(anon=False) # need to manually add key
    else:
        s3 = s3fs.S3FileSystem(anon=False)

    ## create dict for final list of variables and missing dates
    sassie_ecco_variable_gaps = dict()
    
    for variable in variables_list:
        
        ## find filenames for variable
        file_list = np.sort(s3.glob(f'{s3_netcdf_dir}{variable}/*.nc'))
    
        # construct url form of filenames
        # data_urls = ['s3://' + f for f in file_list]

        ## get list of dates from netcdf filenames
        file_dates_list = []
        for netcdf in file_list:
            file_dates_list.append(netcdf.split('/')[-1].split('_')[-6])

        ## identify missing dates
        missing_dates_list = find_missing_dates(date_list_complete, file_dates_list)

        sassie_ecco_variable_gaps[variable] = missing_dates_list

        print(f'{variable} is missing {len(missing_dates_list)}')

    return sassie_ecco_variable_gaps

#### Run function for all variable directories

In [6]:
var_list_all = ["SALT_AVG_DAILY",
                "THETA_AVG_DAILY",
                "UVEL_AVG_DAILY",
                "VVEL_AVG_DAILY",
                "WVEL_AVG_DAILY",
                "KPPdiffS_AVG_DAILY",
                "KPPdiffT_AVG_DAILY",
                "KPPviscA_AVG_DAILY",
                "ETAN_AVG_DAILY",
                "PHIBOT_AVG_DAILY",
                "sIceLoad_AVG_DAILY",
                "KPPhbl_AVG_DAILY",
                "EXFaqh_AVG_DAILY",
                "EXFatemp_AVG_DAILY",
                "EXFempmr_AVG_DAILY",
                "EXFevap_AVG_DAILY",
                "EXFpreci_AVG_DAILY",
                "EXFroff_AVG_DAILY",
                "EXFqnet_AVG_DAILY",
                "EXFhl_AVG_DAILY",
                "EXFhs_AVG_DAILY",
                "EXFlwdn_AVG_DAILY",
                "EXFlwnet_AVG_DAILY",
                "EXFswdn_AVG_DAILY",
                "EXFswnet_AVG_DAILY",
                "EXFuwind_AVG_DAILY",
                "EXFtaux_AVG_DAILY",
                "EXFvwind_AVG_DAILY",
                "EXFtauy_AVG_DAILY",
                "oceFWflx_AVG_DAILY",
                "oceQnet_AVG_DAILY",
                "oceQsw_AVG_DAILY",
                "oceTAUX_AVG_DAILY",
                "oceTAUY_AVG_DAILY",
                "SFLUX_AVG_DAILY",
                "TFLUX_AVG_DAILY",
                "PHIHYD_AVG_DAILY",
                "PHIHYDcR_AVG_DAILY",
                "RHOAnoma_AVG_DAILY",
                "SIatmFW_AVG_DAILY",
                "SIatmQnt_AVG_DAILY",
                "SIarea_AVG_DAILY",
                "SIheff_AVG_DAILY",
                "SIhsnow_AVG_DAILY",
                "SIuice_AVG_DAILY",
                "SIvice_AVG_DAILY",
                "ADVr_SLT_AVG_DAILY",
                "ADVr_TH_AVG_DAILY",
                "ADVxHEFF_AVG_DAILY",
                "ADVxSNOW_AVG_DAILY",
                "ADVyHEFF_AVG_DAILY",
                "ADVySNOW_AVG_DAILY",
                "ADVx_SLT_AVG_DAILY",
                "ADVx_TH_AVG_DAILY",
                "ADVy_SLT_AVG_DAILY",
                "ADVy_TH_AVG_DAILY",
                "DFrE_SLT_AVG_DAILY",
                "DFrE_TH_AVG_DAILY",
                "DFrI_SLT_AVG_DAILY",
                "DFrI_TH_AVG_DAILY",
                "DFxE_SLT_AVG_DAILY",
                "DFxE_TH_AVG_DAILY",
                "DFyE_SLT_AVG_DAILY",
                "DFyE_TH_AVG_DAILY",
                "UVELMASS_AVG_DAILY",
                "VVELMASS_AVG_DAILY",
                "WVELMASS_AVG_DAILY"]

In [7]:
## check number
len(var_list_all)

67

##### First run for ECCO s3 bucket

In [8]:
## s3 directory
ecco_s3_netcdf_dir = 's3://ecco-processed-data/SASSIE/N1/V1/HH/NETCDF/'

In [9]:
ecco_s3_variable_gaps = identify_missing_datasets(var_list_all, ecco_s3_netcdf_dir, sassie_s3 = False)

SALT_AVG_DAILY is missing 2
THETA_AVG_DAILY is missing 2
UVEL_AVG_DAILY is missing 16
VVEL_AVG_DAILY is missing 16
WVEL_AVG_DAILY is missing 16
KPPdiffS_AVG_DAILY is missing 1811
KPPdiffT_AVG_DAILY is missing 1811
KPPviscA_AVG_DAILY is missing 2133
ETAN_AVG_DAILY is missing 16
PHIBOT_AVG_DAILY is missing 16
sIceLoad_AVG_DAILY is missing 16
KPPhbl_AVG_DAILY is missing 134
EXFaqh_AVG_DAILY is missing 2
EXFatemp_AVG_DAILY is missing 2
EXFempmr_AVG_DAILY is missing 2
EXFevap_AVG_DAILY is missing 2
EXFpreci_AVG_DAILY is missing 2
EXFroff_AVG_DAILY is missing 2
EXFqnet_AVG_DAILY is missing 2
EXFhl_AVG_DAILY is missing 2
EXFhs_AVG_DAILY is missing 2
EXFlwdn_AVG_DAILY is missing 2
EXFlwnet_AVG_DAILY is missing 2
EXFswdn_AVG_DAILY is missing 2
EXFswnet_AVG_DAILY is missing 2
EXFuwind_AVG_DAILY is missing 2
EXFtaux_AVG_DAILY is missing 2
EXFvwind_AVG_DAILY is missing 2
EXFtauy_AVG_DAILY is missing 2
oceFWflx_AVG_DAILY is missing 2
oceQnet_AVG_DAILY is missing 2
oceQsw_AVG_DAILY is missing 2
oceT

Save dictionary as json output

In [11]:
# save dictionary as json file
file_name = "sassie-ecco-missing-dates.json"

with open(file_name, 'w') as json_file:
    json.dump(ecco_s3_variable_gaps, json_file)

***

Then run for Fred's s3 bucket

In [97]:
## s3 directory
sassie_s3_netcdf_dir = 's3://podaac-dev-sassie/ECCO_model/N1/V1/HH/NETCDF/'

In [98]:
sassie_s3_variable_gaps = identify_missing_datasets(var_list_all, sassie_s3_netcdf_dir, sassie_s3 = True)

SALT_AVG_DAILY is missing 2
THETA_AVG_DAILY is missing 175
UVEL_AVG_DAILY is missing 16
VVEL_AVG_DAILY is missing 16
WVEL_AVG_DAILY is missing 16
KPPdiffS_AVG_DAILY is missing 1811
KPPdiffT_AVG_DAILY is missing 1811
KPPviscA_AVG_DAILY is missing 2133
ETAN_AVG_DAILY is missing 16
PHIBOT_AVG_DAILY is missing 16
sIceLoad_AVG_DAILY is missing 16
KPPhbl_AVG_DAILY is missing 134
EXFaqh_AVG_DAILY is missing 2
EXFatemp_AVG_DAILY is missing 2
EXFempmr_AVG_DAILY is missing 2
EXFevap_AVG_DAILY is missing 2
EXFpreci_AVG_DAILY is missing 2
EXFroff_AVG_DAILY is missing 2
EXFqnet_AVG_DAILY is missing 2
EXFhl_AVG_DAILY is missing 2
EXFhs_AVG_DAILY is missing 2
EXFlwdn_AVG_DAILY is missing 2
EXFlwnet_AVG_DAILY is missing 2
EXFswdn_AVG_DAILY is missing 2
EXFswnet_AVG_DAILY is missing 2
EXFuwind_AVG_DAILY is missing 2
EXFtaux_AVG_DAILY is missing 2
EXFvwind_AVG_DAILY is missing 2
EXFtauy_AVG_DAILY is missing 2
oceFWflx_AVG_DAILY is missing 2
oceQnet_AVG_DAILY is missing 2
oceQsw_AVG_DAILY is missing 2
oc

In [77]:
## look at one variable as an example
sassie_s3_variable_gaps['ADVySNOW_AVG_DAILY']

['2014-04-24', '2015-04-09']

***

See which files haven't been transferred to Fred's bucket

In [207]:
def get_dates(file_list):
    dates_list = []
    for file in file_list:
        date_tmp = file.split('/')[-1].split('_')[-6]
        dates_list.append(date_tmp)
    return dates_list

In [235]:
def find_mismatch_in_var(file_list_ecco, file_list_sassie):
    set1 = set(file_list_ecco)
    set2 = set(file_list_sassie)
    
    # Elements present in list1 but not in list2
    mismatch_in_lists = list(set1 - set2)
    
    return mismatch_in_lists

In [247]:
def find_mismatch(variable):

    ## get full list of variables for ecco s3
    s3 = s3fs.S3FileSystem(anon=False)
    ecco_s3_netcdf_dir = 's3://ecco-processed-data/SASSIE/N1/V1/HH/NETCDF/'
    file_list_ecco = np.sort(s3.glob(f'{s3_netcdf_dir}{variable}/*.nc'))
    
    ## get full list of variables for sassie s3
    s3 = s3fs.S3FileSystem(anon=False)
    sassie_s3_netcdf_dir = 's3://podaac-dev-sassie/ECCO_model/N1/V1/HH/NETCDF/'
    file_list_sassie = np.sort(s3.glob(f'{sassie_s3_netcdf_dir}{variable}/*.nc'))

    dates_ecco = get_dates(file_list_ecco)
    dates_sassie = get_dates(file_list_sassie)
    missing_dates = find_mismatch_in_var(dates_ecco, dates_sassie)
    
    print(len(missing_dates))
    print(np.sort(missing_dates)[0:10])

    return np.sort(missing_dates)

In [248]:
variable = 'THETA_AVG_DAILY'
theta_mismatch = find_mismatch(variable)

173
['2020-08-15' '2020-08-16' '2020-08-17' '2020-08-18' '2020-08-19'
 '2020-08-20' '2020-08-21' '2020-08-22' '2020-08-23' '2020-08-24']


In [249]:
variable = 'ADVy_SLT_AVG_DAILY'
advy_slt_mismatch = find_mismatch(variable)

424
['2019-09-24' '2019-09-25' '2019-09-26' '2019-09-27' '2019-09-28'
 '2019-09-29' '2019-09-30' '2019-10-01' '2019-10-02' '2019-10-03']


In [252]:
variable = 'VVELMASS_AVG_DAILY'
vvel_mismatch = find_mismatch(variable)

81
['2020-01-24' '2020-01-25' '2020-01-26' '2020-01-27' '2020-01-28'
 '2020-01-29' '2020-01-30' '2020-01-31' '2020-02-01' '2020-02-02']


THETA: 173 files starting from '2020-08-15' to end <br>
ADVy_SLT: 424 files starting from '2019-09-24' to end <br>
VVELMASS: 81 files from '2020-01-24' to '2020-09-28' <br>

##### Compare datasets between Ian's and Fred's

In [151]:
for variable in var_list_all:
    ## subtract number of files missing in ecco bucket from sassie bucket
    diff_tmp = len(sassie_s3_variable_gaps[variable]) - len(ecco_s3_variable_gaps[variable])
    
    print(f'{variable}: {diff_tmp}')

SALT_AVG_DAILY: 0
THETA_AVG_DAILY: 173
UVEL_AVG_DAILY: 0
VVEL_AVG_DAILY: 0
WVEL_AVG_DAILY: 0
KPPdiffS_AVG_DAILY: 0
KPPdiffT_AVG_DAILY: 0
KPPviscA_AVG_DAILY: 0
ETAN_AVG_DAILY: 0
PHIBOT_AVG_DAILY: 0
sIceLoad_AVG_DAILY: 0
KPPhbl_AVG_DAILY: 0
EXFaqh_AVG_DAILY: 0
EXFatemp_AVG_DAILY: 0
EXFempmr_AVG_DAILY: 0
EXFevap_AVG_DAILY: 0
EXFpreci_AVG_DAILY: 0
EXFroff_AVG_DAILY: 0
EXFqnet_AVG_DAILY: 0
EXFhl_AVG_DAILY: 0
EXFhs_AVG_DAILY: 0
EXFlwdn_AVG_DAILY: 0
EXFlwnet_AVG_DAILY: 0
EXFswdn_AVG_DAILY: 0
EXFswnet_AVG_DAILY: 0
EXFuwind_AVG_DAILY: 0
EXFtaux_AVG_DAILY: 0
EXFvwind_AVG_DAILY: 0
EXFtauy_AVG_DAILY: 0
oceFWflx_AVG_DAILY: 0
oceQnet_AVG_DAILY: 0
oceQsw_AVG_DAILY: 0
oceTAUX_AVG_DAILY: 0
oceTAUY_AVG_DAILY: 0
SFLUX_AVG_DAILY: 0
TFLUX_AVG_DAILY: 0
PHIHYD_AVG_DAILY: 0
PHIHYDcR_AVG_DAILY: 0
RHOAnoma_AVG_DAILY: 0
SIatmFW_AVG_DAILY: 0
SIatmQnt_AVG_DAILY: 0
SIarea_AVG_DAILY: 0
SIheff_AVG_DAILY: 0
SIhsnow_AVG_DAILY: 0
SIuice_AVG_DAILY: 0
SIvice_AVG_DAILY: 0
ADVr_SLT_AVG_DAILY: 0
ADVr_TH_AVG_DAILY: 0
ADVxHEFF

***

Identify which variables are empty

In [11]:
def find_variable_with_large_list(dictionary):
    keys_with_gaps = []
    
    # Iterate through the dictionary
    for key, value in dictionary.items():
        # Check if the length of the list is more than XX dates
        if len(value) > 2580:
            keys_with_gaps.append(key)
    
    return keys_with_gaps

In [12]:
variables_with_large_gaps = find_variable_with_large_list(sassie_ecco_variable_gaps)

In [13]:
variables_with_large_gaps

['ADVr_SLT_AVG_DAILY',
 'ADVr_TH_AVG_DAILY',
 'ADVx_SLT_AVG_DAILY',
 'ADVx_TH_AVG_DAILY',
 'ADVy_SLT_AVG_DAILY',
 'ADVy_TH_AVG_DAILY',
 'DFrE_SLT_AVG_DAILY',
 'DFrE_TH_AVG_DAILY',
 'DFrI_SLT_AVG_DAILY',
 'DFrI_TH_AVG_DAILY',
 'DFxE_SLT_AVG_DAILY',
 'DFxE_TH_AVG_DAILY',
 'DFyE_SLT_AVG_DAILY',
 'DFyE_TH_AVG_DAILY']

### BOOKEEPING NOTES

63 datasets on s3 bucket now (identify missing 4)

9 items that need to be added to sassie cloud:

In [ ]:
"ADVx_TH_AVG_DAILY"
"EXFuwind_AVG_DAILY"
"KPPdiffT_AVG_DAILY"
"KPPhbl_AVG_DAILY"
"oceFWflx_AVG_DAILY"
"SFLUX_AVG_DAILY"
"SIatmQnt_AVG_DAILY"
"SIuice_AVG_DAILY"
"VVEL_AVG_DAILY"

What's on sassie cloud now (54)

In [73]:
var_list = ["ADVr_SLT_AVG_DAILY",
"ADVr_TH_AVG_DAILY",
"ADVxHEFF_AVG_DAILY",
"ADVxSNOW_AVG_DAILY",
"ADVx_SLT_AVG_DAILY",
"ADVyHEFF_AVG_DAILY",
"ADVySNOW_AVG_DAILY",
"ADVy_SLT_AVG_DAILY",
"ADVy_TH_AVG_DAILY",
"DFrE_SLT_AVG_DAILY",
"DFrE_TH_AVG_DAILY",
"DFrI_SLT_AVG_DAILY",
"DFrI_TH_AVG_DAILY",
"ETAN_AVG_DAILY",
"EXFaqh_AVG_DAILY",
"EXFatemp_AVG_DAILY",
"EXFempmr_AVG_DAILY",
"EXFevap_AVG_DAILY",
"EXFhl_AVG_DAILY",
"EXFhs_AVG_DAILY",
"EXFlwdn_AVG_DAILY",
"EXFlwnet_AVG_DAILY",
"EXFpreci_AVG_DAILY",
"EXFqnet_AVG_DAILY",
"EXFroff_AVG_DAILY",
"EXFswdn_AVG_DAILY",
"EXFswnet_AVG_DAILY",
"EXFtaux_AVG_DAILY",
"EXFtauy_AVG_DAILY",
"EXFvwind_AVG_DAILY",
"KPPdiffS_AVG_DAILY",
"KPPviscA_AVG_DAILY",
"PHIBOT_AVG_DAILY",
"PHIHYD_AVG_DAILY",
"PHIHYDcR_AVG_DAILY",
"RHOAnoma_AVG_DAILY",
"SALT_AVG_DAILY",
"SIarea_AVG_DAILY",
"SIatmFW_AVG_DAILY",
"SIheff_AVG_DAILY",
"SIhsnow_AVG_DAILY",
"SIvice_AVG_DAILY",
"TFLUX_AVG_DAILY",
"THETA_AVG_DAILY",
"UVELMASS_AVG_DAILY",
"UVEL_AVG_DAILY",
"VVELMASS_AVG_DAILY",
"WVELMASS_AVG_DAILY",
"WVEL_AVG_DAILY",
"oceQnet_AVG_DAILY",
"oceQsw_AVG_DAILY",
"oceTAUX_AVG_DAILY",
"oceTAUY_AVG_DAILY",
"sIceLoad_AVG_DAILY"]

In [75]:
len(var_list)

54

In [78]:
var_list[0:32]

['ADVr_SLT_AVG_DAILY',
 'ADVr_TH_AVG_DAILY',
 'ADVxHEFF_AVG_DAILY',
 'ADVxSNOW_AVG_DAILY',
 'ADVx_SLT_AVG_DAILY',
 'ADVyHEFF_AVG_DAILY',
 'ADVySNOW_AVG_DAILY',
 'ADVy_SLT_AVG_DAILY',
 'ADVy_TH_AVG_DAILY',
 'DFrE_SLT_AVG_DAILY',
 'DFrE_TH_AVG_DAILY',
 'DFrI_SLT_AVG_DAILY',
 'DFrI_TH_AVG_DAILY',
 'ETAN_AVG_DAILY',
 'EXFaqh_AVG_DAILY',
 'EXFatemp_AVG_DAILY',
 'EXFempmr_AVG_DAILY',
 'EXFevap_AVG_DAILY',
 'EXFhl_AVG_DAILY',
 'EXFhs_AVG_DAILY',
 'EXFlwdn_AVG_DAILY',
 'EXFlwnet_AVG_DAILY',
 'EXFpreci_AVG_DAILY',
 'EXFqnet_AVG_DAILY',
 'EXFroff_AVG_DAILY',
 'EXFswdn_AVG_DAILY',
 'EXFswnet_AVG_DAILY',
 'EXFtaux_AVG_DAILY',
 'EXFtauy_AVG_DAILY',
 'EXFvwind_AVG_DAILY',
 'KPPdiffS_AVG_DAILY',
 'KPPviscA_AVG_DAILY']

### Look at 2D fields in sassie s3

In [ ]:
'ETAN_AVG_DAILY'
'PHIBOT_AVG_DAILY'
'sIceLoad_AVG_DAILY'
'KPPhbl_AVG_DAILY'
'EXFaqh_AVG_DAILY'
'EXFatemp_AVG_DAILY'
'EXFempmr_AVG_DAILY'
'EXFevap_AVG_DAILY'
'EXFpreci_AVG_DAILY'
'EXFroff_AVG_DAILY'
'EXFqnet_AVG_DAILY'
'EXFhl_AVG_DAILY'
'EXFhs_AVG_DAILY'
'EXFlwdn_AVG_DAILY'
'EXFlwnet_AVG_DAILY'
'EXFswdn_AVG_DAILY'
'EXFswnet_AVG_DAILY'
'EXFuwind_AVG_DAILY'
'EXFtaux_AVG_DAILY'
'EXFvwind_AVG_DAILY'
'EXFtauy_AVG_DAILY'
'oceFWflx_AVG_DAILY'
'oceQnet_AVG_DAILY'
'oceQsw_AVG_DAILY'
'oceTAUX_AVG_DAILY'
'oceTAUY_AVG_DAILY'
'SFLUX_AVG_DAILY'
'TFLUX_AVG_DAILY'
'SIatmFW_AVG_DAILY'
'SIatmQnt_AVG_DAILY'
'SIarea_AVG_DAILY'
'SIheff_AVG_DAILY'
'SIhsnow_AVG_DAILY'
'SIuice_AVG_DAILY'
'SIvice_AVG_DAILY'
'ADVxHEFF_AVG_DAILY'
'ADVxSNOW_AVG_DAILY'
'ADVyHEFF_AVG_DAILY'
'ADVySNOW_AVG_DAILY'

In [86]:
fields_2d = [
    "ETAN_AVG_DAILY",
    "PHIBOT_AVG_DAILY",
    "sIceLoad_AVG_DAILY",
    "KPPhbl_AVG_DAILY",
    "EXFaqh_AVG_DAILY",
    "EXFatemp_AVG_DAILY",
    "EXFempmr_AVG_DAILY",
    "EXFevap_AVG_DAILY",
    "EXFpreci_AVG_DAILY",
    "EXFroff_AVG_DAILY",
    "EXFqnet_AVG_DAILY",
    "EXFhl_AVG_DAILY",
    "EXFhs_AVG_DAILY",
    "EXFlwdn_AVG_DAILY",
    "EXFlwnet_AVG_DAILY",
    "EXFswdn_AVG_DAILY",
    "EXFswnet_AVG_DAILY",
    "EXFuwind_AVG_DAILY",
    "EXFtaux_AVG_DAILY",
    "EXFvwind_AVG_DAILY",
    "EXFtauy_AVG_DAILY",
    "oceFWflx_AVG_DAILY",
    "oceQnet_AVG_DAILY",
    "oceQsw_AVG_DAILY",
    "oceTAUX_AVG_DAILY",
    "oceTAUY_AVG_DAILY",
    "SFLUX_AVG_DAILY",
    "TFLUX_AVG_DAILY",
    "SIatmFW_AVG_DAILY",
    "SIatmQnt_AVG_DAILY",
    "SIarea_AVG_DAILY",
    "SIheff_AVG_DAILY",
    "SIhsnow_AVG_DAILY",
    "SIuice_AVG_DAILY",
    "SIvice_AVG_DAILY",
    "ADVxHEFF_AVG_DAILY",
    "ADVxSNOW_AVG_DAILY",
    "ADVyHEFF_AVG_DAILY",
    "ADVySNOW_AVG_DAILY"
]

In [87]:
## s3 directory
sassie_s3_netcdf_dir = 's3://podaac-dev-sassie/ECCO_model/N1/V1/HH/NETCDF/'

In [89]:
sassie_s3_2D_variable_gaps = identify_missing_datasets(fields_2d, sassie_s3_netcdf_dir, sassie_s3 = True)

ETAN_AVG_DAILY is missing 16
PHIBOT_AVG_DAILY is missing 16
sIceLoad_AVG_DAILY is missing 16
KPPhbl_AVG_DAILY is missing 134
EXFaqh_AVG_DAILY is missing 2
EXFatemp_AVG_DAILY is missing 2
EXFempmr_AVG_DAILY is missing 2
EXFevap_AVG_DAILY is missing 2
EXFpreci_AVG_DAILY is missing 2
EXFroff_AVG_DAILY is missing 2
EXFqnet_AVG_DAILY is missing 2
EXFhl_AVG_DAILY is missing 2
EXFhs_AVG_DAILY is missing 2
EXFlwdn_AVG_DAILY is missing 2
EXFlwnet_AVG_DAILY is missing 2
EXFswdn_AVG_DAILY is missing 2
EXFswnet_AVG_DAILY is missing 2
EXFuwind_AVG_DAILY is missing 2
EXFtaux_AVG_DAILY is missing 2
EXFvwind_AVG_DAILY is missing 2
EXFtauy_AVG_DAILY is missing 2
oceFWflx_AVG_DAILY is missing 2
oceQnet_AVG_DAILY is missing 2
oceQsw_AVG_DAILY is missing 2
oceTAUX_AVG_DAILY is missing 2
oceTAUY_AVG_DAILY is missing 2
SFLUX_AVG_DAILY is missing 2
TFLUX_AVG_DAILY is missing 2
SIatmFW_AVG_DAILY is missing 2
SIatmQnt_AVG_DAILY is missing 2
SIarea_AVG_DAILY is missing 2
SIheff_AVG_DAILY is missing 2
SIhsnow_AV

### Look at 2D and 3D fields

In [84]:
fields_2d = [
    "ETAN",
    "PHIBOT",
    "sIceLoad",
    "KPPhbl",
    "EXFaqh",
    "EXFatemp",
    "EXFempmr",
    "EXFevap",
    "EXFpreci",
    "EXFroff",
    "EXFqnet",
    "EXFhl",
    "EXFhs",
    "EXFlwdn",
    "EXFlwnet",
    "EXFswdn",
    "EXFswnet",
    "EXFuwind",
    "EXFtaux",
    "EXFvwind",
    "EXFtauy",
    "oceFWflx",
    "oceQnet",
    "oceQsw",
    "oceTAUX",
    "oceTAUY",
    "SFLUX",
    "TFLUX",
    "SIatmFW",
    "SIatmQnt",
    "SIarea",
    "SIheff",
    "SIhsnow",
    "SIuice",
    "SIvice",
    "ADVxHEFF",
    "ADVxSNOW",
    "ADVyHEFF",
    "ADVySNOW"
]

In [ ]:
fields_3d = [
    "SALT",
    "THETA",
    "UVEL",
    "VVEL",
    "WVEL",
    "KPPdiffS",
    "KPPdiffT",
    "KPPviscA",
    "PHIHYD",
    "PHIHYDcR",
    "RHOAnoma",
    "ADVr_SLT",
    "ADVr_TH",
    "ADVx_SLT",
    "ADVx_TH",
    "ADVy_SLT",
    "ADVy_TH",
    "DFrE_SLT",
    "DFrE_TH",
    "DFrI_SLT",
    "DFrI_TH",
    "DFxE_SLT",
    "DFxE_TH",
    "DFyE_SLT",
    "DFyE_TH",
    "UVELMASS",
    "VVELMASS",
    "WVELMASS"
]